File: ios/RoutineChart/RoutineChart/Core/Domain/Models/Family.swift
//
//  Family.swift
//  RoutineChart
//
//  Domain model for Family entity
//

import Foundation

struct Family: Identifiable, Codable, Equatable, Hashable {
    let id: String
    var name: String?
    var timeZone: String
    var weekStartsOn: Int
    var planTier: PlanTier
    let createdAt: Date
    var updatedAt: Date
    
    init(
        id: String = UUID().uuidString,
        name: String? = nil,
        timeZone: String = TimeZone.current.identifier,
        weekStartsOn: Int = 0,
        planTier: PlanTier = .free,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.name = name
        self.timeZone = timeZone
        self.weekStartsOn = weekStartsOn
        self.planTier = planTier
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/models/Family.kt
package com.HammersTech.RoutineChart.core.domain.models

import java.time.Instant

/**
 * Family domain model
 * Represents a household unit with shared routines
 */
data class Family(
    val id: String,
    val name: String?,
    val timeZone: String,
    val weekStartsOn: Int,
    val planTier: PlanTier,
    val createdAt: Instant,
    val updatedAt: Instant
) {
    companion object {
        const val DEFAULT_TIMEZONE = "America/Los_Angeles"
        const val DEFAULT_WEEK_STARTS_ON = 0 // Sunday
    }
}

